unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,27 @@
+#lang racket/base
+
+(provide hash-pluck-lens)
+
+(require racket/list
+         lens/base/main
+         "hash.rkt"
+         "compound.rkt")
+(module+ test
+  (require rackunit))
+
+(define (hash-pluck-lens . ks)
+  (apply compound-hash-lens
+         (append-map
+          (λ (k)
+            (list k (hash-ref-lens k)))
+          ks)))
+
+(module+ test
+  (check-equal? (lens-view (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4 'c 5))
+                (hash 'a 4 'b 2 'c 5))
+  (test-case "this one currently fails"
+    (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4))
+                  (hash 'a 4 'b 2 'c 3)))
```

Comment:
To skirt around the problem completely, we could just leave setting smaller hashes as an exception and if we later decide that we want that feature figure out how to implement it then.